lessons/04-structured-output/README.md

```diff
@@ -0,0 +1,590 @@
+> Board and Pieces: The game is played on an X-shaped board with a center block and four arms. Players use five black beans as dice, each marked on one side 
+> 
+> Setup: Each player selects six markers (beads) of their color and places them off the board. Players also have point beads for scoring 
+> 
+> Starting the Game: Players roll the beans to move their markers around the board. The goal is to move all markers around the board and back to the starting point 
+> 
+> Movement: Players roll the beans to determine how many spaces to move. A roll of five allows a marker to move ten spaces 
+> 
+> Capturing and Scoring: Landing on specific spaces can result in gaining extra turns, losing point beads, or capturing opponent's markers 
+> 
+> Aztec Involvement: 
+> 
+> The Aztecs, including nobles and commoners, widely played Patolli. It was a popular pastime at the court of Montezuma, who enjoyed watching his nobles play 
+> 
+> The game was not just for entertainment but also involved high-stakes gambling, with players betting items like blankets, precious stones, and even their homes or freedom 
+
+![Patolli board](./assets/patolli.png)
+
+*Patolli - Wikipedia*
+
+## Structured Output
+
+**Time beetle:** Hey folks, want to know how I beat Montezuma?
+
+**You:** Sshh, not so loud, we're still in the Aztec Empire.
+
+**Time beetle:** Oh right, sorry, yea so structured output is how. You asked for the right moves and I gave them to you in a structured way.
+
+Structured output refers to different data formats that are used to present information in a structured manner. Formats like JSON, XML, and CSV are commonly used and makes it easy to consume data by services.
+
+Generative AI models can output data in various formats. However, output of unstructured text can make it difficult to extract information and hand it over to other services. Structured output can help in this regard.
+
+**You:** Well, aren't you clever ;)
+
+**Time beetle:** I try my best :)
+
+Another aspect, even if for human consumption is that it's easier to read and understand the data if it follows a structure that is familiar to the reader.
+
+Let's look at a few examples of input and asking for specific output formats can help in extracting information that at least makes it easier to understand and consume.
+
+**You:** Ok, it's not like we got any other plans, let's hear it.
+
+## Shaping the output
+
+**Time beetle:** Let's start with a simple example. Imagine you're Leonardo da Vinci and you're trying to come up with ideas for inventions using 15th-century technology. You want to know what resources you need and how much effort it would take to build each invention. Here's how you can ask for this information:
+
+**Leonardo:** Oh I like this game, everyone pretend you're me :)
+
+> Prompt: Generate ideas for inventions I can build given 1500th century technology, please let me know for each what resources I need and how much effort there would be to build each invention
```

Comment:
Prompt: Generate ideas for inventions I can build using 15th-century technology. For each invention, specify the required resources and estimate the effort needed to build it.